### PR TITLE
Torch: `map_binary_to_indices`, `map_classes_to_indices`, `correct_crop_centers`, `generate_pos_neg_label_crop_centers`, `generate_label_classes_crop_centers`

### PR DESCRIPTION
**Pull request by rijobro**
### Description
Torch: `map_binary_to_indices`, `map_classes_to_indices`, `correct_crop_centers`, `generate_pos_neg_label_crop_centers`, `generate_label_classes_crop_centers`

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by wyli — APPROVED** [review]
looks good to me, @Nic-Ma this might change the behaviour of the spleen seg. notebook...please confirm

File: monai/transforms/croppad/array.py [hunk 1]
```diff
@@ -910,16 +910,18 @@ def randomize(
         image: Optional[np.ndarray] = None,
     ) -> None:
         self.spatial_size = fall_back_tuple(self.spatial_size, default=label.shape[1:])
+        fg_indices_: np.ndarray
```
Comment (Nic-Ma) [inline monai/transforms/croppad/array.py:913]: May I know why the data type is still `numpy array`? I think you added `Tensor` support?

Thanks.

**Review by Nic-Ma — APPROVED** [review]
Looks good to me.
Could you please help double confirm by executing this example?
https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/torch/unet_training_dict.py
Just also need to add `EnsureTyped()` after `LoadImaged()` to test with Tensor data.

Thanks in advance.

File: monai/transforms/croppad/array.py (hunk 1 above)
Comment (rijobro) [reply monai/transforms/croppad/array.py:913]: I've updated the helper functions, but not actually updated any of the transforms. Hence, the transforms expect numpy input, and therefore expect numpy output from these helper functions. This PR will then enable me to start updating the transforms that use the helper functions.

Comment (rijobro) [reply monai/transforms/croppad/array.py:913]: @Nic-Ma thanks for the suggestion, the example script now runs both normally and with `EnsureTyped` after `LoadImaged`.

Comment (Nic-Ma) [reply monai/transforms/croppad/array.py:913]: I see your point, thanks!

**Comment by Nic-Ma** [comment]
Hi @rijobro ,

Would you like to add Tensor support for `RandCropByPosNegLabel` and `RandCropByLabelClasses` with a high priority? As you already updated the help utilities.
And I think this PosNegCrop transform is popular in our training tasks, if we can support Tensor in v0.7, that would be great.

Thanks in advance.

**Pull request merged**